UefiPayloadPkg/UniversalPayloadBuild.py

```diff
@@ -130,10 +130,10 @@ def BuildUniversalPayload(Args):
     BuildDir     = os.path.join(os.environ['WORKSPACE'], os.path.normpath("Build/UefiPayloadPkg{}").format (Args.Arch))
     if Args.Arch == 'X64':
         BuildArch      = "X64"
-        FitArch        = "x86_64"
+        FitArch        = "i386"

```

Comment:
No, this is a coreboot-specific point. *coreboot* starts its payloads in 32-bit on x86, without condition on what coreboot is compiled as. So, coreboot will build UPL passing in `-a IA32`. EDK2-based Platform Init might call UPL already in long mode, or it might do a jump from their 32-bit PEI, to our 32-bit SEC, and let us do the mode switch.

U-Boot should build with `-a IA32` or `-a X64` depending on the bitness of the instructions right before you jump.